### PR TITLE
Provider gui as status checker for providers on contract

### PR DESCRIPTION
**Pull request by HughParry**

### CODE REVIEW

File: provider-gui/contexts/GlobalContext.tsx [hunk 1]
```diff
@@ -5,11 +5,15 @@ import React, { ReactNode, createContext, useContext, useState } from 'react'
 interface GlobalStateContextProps {
     currentAccount: string
     setCurrentAccount: (updateStr: string) => void
+    network: 'rococo' | 'development'
+    setNetwork: (updateStr: 'rococo' | 'development') => void
```
Comment (goastler) [inline provider-gui/contexts/GlobalContext.tsx:9]: these strs should probbaly come from the config pkg. Don't have to do it right now, but could you add a ticket for it?

File: provider-gui/services/contract/contractOverview.ts [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+import { GovernanceStatus } from '@prosopo/captcha-contract'
+import { ProcaptchaClientConfigOutput } from '@prosopo/types'
+import { ProsopoCaptchaContract } from '@prosopo/contract'
+import { ContractAbi as abiJson } from '@prosopo/captcha-contract'
+import { hexToString } from '@polkadot/util'
+
+const deployedContracts = {
+    rococo: {
+        wsProviderUrl: 'wss://rococo-contracts-rpc.polkadot.io:443',
+        contracts: ['5HiVWQhJrysNcFNEWf2crArKht16zrhro3FcekVWocyQjx5u'],
```
Comment (goastler) [inline provider-gui/services/contract/contractOverview.ts:11]: is there any ability to enter our own contract here? I'm imaging this tool being hosted somewhere and I've just deployed a contract, don't want to recompile and restart the provider status tool so would be easier just to type in the contract address manually. Don't have to add now tho, could just put it into a ticket

File: provider-gui/services/contract/contractOverview.ts [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+    },
+    development: { wsProviderUrl: 'ws://localhost:9944', contracts: [] },
```
Comment (goastler) [inline provider-gui/services/contract/contractOverview.ts:13]: likewise with networks, would be good to be able to add our own somewhat easily so I can reuse this tool on astar, etc, without having to recompile

File: provider-gui/types/ContractOverview.ts [hunk 4]
```diff
@@ -0,0 +1,15 @@
+export interface ContractOverview {
+    contractAddress: string
+    network: 'development' | 'rococo'
```
Comment (goastler) [inline provider-gui/types/ContractOverview.ts:3]: should be populated from some config somewhere, same as above

File: provider-gui/components/AccountPicker.tsx [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+    accounts: web3Accounts,
+}
+
+const AccountPicker: React.FC = () => {
+    const [accounts, setAccounts] = useState<InjectedAccountWithMeta[]>([])
+    const { currentAccount, setCurrentAccount, network, setNetwork } = useGlobalState()
+
+    useEffect(() => {
+        const enableExtension = async () => {
+            await injected.enable('my cool dapp')
```
Comment (goastler) [inline provider-gui/components/AccountPicker.tsx]: "my cool dapp" should probs be changed :joy:

File: provider-gui/components/AccountPicker.tsx [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+import { useGlobalState } from '../contexts/GlobalContext'
+import { web3Accounts, web3Enable } from '@polkadot/extension-dapp'
+import React from 'react'
+
+interface Injected {
+    enable: typeof web3Enable
+    accounts: typeof web3Accounts
+}
+
+const injected: Injected = {
```
Comment (goastler) [inline provider-gui/components/AccountPicker.tsx]: is this needed? can't you access web3Enable and web3Accounts directly without this wrapper?

**Review by goastler — CHANGES_REQUESTED** [review]

File: provider-gui/contexts/GlobalContext.tsx (hunk 1 above)
Comment (HughParry) [reply provider-gui/contexts/GlobalContext.tsx:9]: done

File: provider-gui/services/contract/contractOverview.ts (hunk 3 above)
Comment (HughParry) [reply provider-gui/services/contract/contractOverview.ts:13]: Ticket in

File: provider-gui/services/contract/contractOverview.ts (hunk 2 above)
Comment (HughParry) [reply provider-gui/services/contract/contractOverview.ts:11]: https://github.com/prosopo/captcha/issues/771

File: provider-gui/types/ContractOverview.ts (hunk 4 above)
Comment (HughParry) [reply provider-gui/types/ContractOverview.ts:3]: Ticket in

File: provider-gui/components/AccountPicker.tsx (hunk 5 above)
Comment (HughParry) [reply provider-gui/components/AccountPicker.tsx]: Changed to Prosopo Provider GUI

File: provider-gui/components/AccountPicker.tsx (hunk 6 above)
Comment (HughParry) [reply provider-gui/components/AccountPicker.tsx]: Done

**Review by goastler — APPROVED** [review]
:rocket:

**Pull request merged**